webapp/utils/subgraph.ts

```diff
@@ -97,15 +115,13 @@ export const getLastIndexedBlock = function (chainId: Chain['id']) {
       }
     }`,
   }
-  return fetch(url, {
-    body: JSON.stringify(schema),
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    method: 'POST',
-  }).then(
-    (r: GraphResponse<{ _meta: { block: { number: number } } }>) =>
-      r.data._meta.block.number,
+  return request<GraphResponse<{ _meta: { block: { number: number } } }>>(
+    url,
+    schema,
+  ).then(
+    response => (
+      checkGraphQLErrors(response), response.data._meta.block.number
+    ),
```

Comment:
As suggested in the previous comments, I think this format makes it harder to read. Let's use classic functions with `return`

```suggestion
  ).then(
    function (response) {
      checkGraphQLErrors(response)
      return response.data._meta.block.number
    },
```

Apply this to all the changes, other than that, LGTM

Reply:
@gndelia all done! Sorry i misunderstood the expected change request. Fixed now

Reply:
Looks great! Thanks!